### PR TITLE
added socketio channel implementation

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- added socketio channel implementation

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/channels/socketio.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    def from_credentials(cls, credentials):
+        credentials = credentials or {}
+        return cls(credentials.get("user_message_evt", "user_uttered"),
+                   credentials.get("bot_message_evt", "bot_uttered"),
+                   credentials.get("namespace"))
+
+    def __init__(self,
+                 user_message_evt="user_uttered",  # type: Text
+                 bot_message_evt="bot_uttered",  # type: Text
+                 namespace=None  # type: Optional[Text]
```
Comment (akelad) [inline rasa_core/channels/socketio.py]: what's this namespace for? if we're reading it from the credentials file it should also be explained in the docs right?

File: rasa_core/channels/socketio.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        @socketio_webhook.route("/", methods=['GET'])
+        def health():
+            return jsonify({"status": "ok"})
+
+        @sio.on('connect', namespace=self.namespace)
+        def connect(sid, environ):
+            logger.debug("User {} connected to socketio endpoint.".format(sid))
+
+        @sio.on('disconnect', namespace=self.name())
```
Comment (akelad) [inline rasa_core/channels/socketio.py]: why is this `self.name()` and the `connect` one  above `self.namespace`?

File: rasa_core/channels/socketio.py (hunk 1 above)
Comment (tmbo) [reply rasa_core/channels/socketio.py]: no clue :D didn't read the socket io docs - but i think it allows you to use the same message names in different contexts. don't think it is important though.

File: rasa_core/channels/socketio.py (hunk 2 above)
Comment (tmbo) [reply rasa_core/channels/socketio.py]: good spot, should be namespace as well :+1:

**Pull request merged**